Url dependencies refactor (#260)
fossas/spectrometer: docs/userguide.md

- name: foo
  version: 1.2.3
  license: "MIT or Apache-2.0"

> # You can also provide a description and/or homepage

Reviewer: ```suggestion
# You can also provide a description and/or homepage. These values populate metadata fields in reports in the FOSSA web UI.
```

- name: foo-wrapper
  version: 1.2.3
  license: MIT
  metadata:
    homepage: https://www.foowrapper.com/about
    description: Provides foo and a helpful interface around foo-like tasks.
```

### Remote dependencies

> FOSSA also supports dependencies that can't be automatically discovered or identified, but the user has a remote URL that points to an archive where the dependency is hosted.

Reviewer: ```suggestion
FOSSA also supports dependencies that can't be automatically discovered or identified, but where the user has a URL where FOSSA can download the source code of the dependency.
```

To do this, you must supply the name, version, and url of the dependency. The FOSSA backend will attempt to download and scan the archive hosted at this remote location. Example for a github source dependency: `https://github.com/fossas/spectrometer/archive/refs/tags/v2.7.2.tar.gz`. You may also supply a description that will be included inside your report, however this is optional.

```yaml

> remote-dependencies:

Reviewer 2: Let's also add this section to the JSON example

  description: Provides foo and a helpful interface around foo-like tasks.
```

### Remote dependencies

FOSSA also supports dependencies that can't be automatically discovered or identified, but the user has a remote URL that points to an archive where the dependency is hosted.

> To do this, you must supply the name, version, and url of the dependency. The FOSSA backend will attempt to download and scan the archive hosted at this remote location. Example for a github source dependency: `https://github.com/fossas/spectrometer/archive/refs/tags/v2.7.2.tar.gz`. You may also supply a description that will be included inside your report, however this is optional.

Reviewer 3: Is version required by the backend? I think allowing users to provide the version is great to reduce scan time, but if the contents at the URL change we won't be picking up any of those changes. 

Due to the increase in scan time to hash the downloaded file, I think there's a chance this is a limitation we're ok with, but we should make this very clear to users so that they don't try to use this for that purpose.

Example: Company keeps the latest version at a URL that doesn't change (https://fossa.com/fossa-release/latest/fossa.tar) but they upload the latest version of the dependency to that URL every time a new version is released.

Author: It will be required, yes. I think what you're describing will be an edge case that we're ok with

Reviewer: (Sorry to necro-post this thread, but GitHub won't let me start a new comment on this line for some reason.)

```suggestion
To specify a remote dependency, you must provide the name, version, and download URL of the dependency. The FOSSA backend will attempt to download and scan any source code contained in an archive hosted at this URL.

For example, for a dependency released on a GitHub release, your URL might look like: `https://github.com/fossas/spectrometer/archive/refs/tags/v2.7.2.tar.gz`.

You can also optionally add metadata fields ("description" and "homepage") to populate these fields in the FOSSA web UI (these fields can be displayed when generating reports).
```
